src/V3EmitMk.cpp

```diff
@@ -31,11 +31,557 @@ class EmitMk final {
 public:
     // METHODS
 
+    struct FileOrConcatenatedFilesList final {
```

Comment:
Please move this into a top-level class instead of inside EmitMk, so the coupling is more explicit and easier to read.  It also might be large enough and unrelated to Mk to deserve a new .h/.cpp file.

Reply:
Reminder ;)

Reply:
Reminder: Please move this into a top-level class instead of inside EmitMk, so the coupling is more explicit and easier to read. It also might be large enough and unrelated to Mk to deserve a new .h/.cpp file.

Reply:
Moved to a top-level class. I am not sure if it deserves a separate .cpp file; this new logic is not very related to emitting Makefiles directly, and is a few more lines long than all the rest of the file, but combined still fits under a thousand lines. Moving does seem reasonable, although for now I decided to keep it here.